service/worker/workerdeployment/version_workflow.go

```diff
@@ -233,6 +236,14 @@ func (d *VersionWorkflowRunner) handleUpdateVersionMetadata(ctx workflow.Context
 }
 
 func (d *VersionWorkflowRunner) startDrainage(ctx workflow.Context, isCan bool) {
+	if d.VersionState.GetDrainageInfo().GetStatus() == enumspb.VERSION_DRAINAGE_STATUS_UNSPECIFIED {
+		now := timestamppb.New(workflow.Now(ctx))
+		d.VersionState.DrainageInfo = &deploymentpb.VersionDrainageInfo{
```

Comment:
hopefully no logic depends on `VersionState.DrainageInfo` otherwise we have to patch.

Reply:
I remember making sure that everything did `VersionState.GetDrainageInfo().GetStatus()` so it /should/ just say unspecified if it's nil due to the gap between child starting and signal arriving

Reply:
I mean workflow patch for NDE

Reply:
I added a patch because I'm adding a signal via syncSummary